### PR TITLE
Add a flag as an option to fix naked returns

### PR DESCRIPTION
**Pull request by joshprzybyszewski-wf**
When discovering a lot of naked returns in a codebase, it'd be nice to just have a script to update them all. This PR attempts to be that script.

It also attempts to clean up some of the file parsing by removing a copied file that said was no longer needed after https://golang.org/issue/8768 is resolved -- which it has been.

Reviewers: @alexkohler

### CODE REVIEW

File: nakedret.go [hunk 1]
```diff
@@ -42,129 +48,97 @@ func main() {
...
+	shouldFix := flag.Bool("fix", false, "whether or not the tool should fix the naked returns")
 	flag.Usage = usage
 	flag.Parse()
 
-	if err := checkNakedReturns(flag.Args(), maxLength); err != nil {
-		log.Println(err)
+	err := doEverything(flag.Args(), *maxLength, *shouldFix)
+	if err != nil {
+		log.Fatalf("Encountered an error: %+v", err)
+		os.Exit(1)
```
Comment (alexkohler) [inline nakedret.go]: As per https://golang.org/pkg/log/#Fatalf, no need for the `os.Exit(1)` here:

> Fatalf is equivalent to Printf() followed by a call to os.Exit(1).

File: nakedret.go [hunk 2]
```diff
@@ -42,129 +48,97 @@ func main() {
...
-
-	fset := token.NewFileSet()
-
-	files, err := parseInput(args, fset)
-	if err != nil {
-		return fmt.Errorf("could not parse input %v", err)
+func doEverything(args []string, maxLength uint, shouldFix bool) error {
+	if len(args) == 0 {
+		// We're just going to check for the current directory
+		checkRequestedFiles(``, maxLength, shouldFix)
```
Comment (alexkohler) [inline nakedret.go]: Could we change this "``" to ""?

File: nakedret.go [hunk 3]
```diff
@@ -42,129 +48,97 @@ func main() {
 	log.SetFlags(0)
 
 	maxLength := flag.Uint("l", 5, "maximum number of lines for a naked return function")
+	shouldFix := flag.Bool("fix", false, "whether or not the tool should fix the naked returns")
 	flag.Usage = usage
 	flag.Parse()
 
-	if err := checkNakedReturns(flag.Args(), maxLength); err != nil {
-		log.Println(err)
+	err := doEverything(flag.Args(), *maxLength, *shouldFix)
```
Comment (alexkohler) [inline nakedret.go]: Keep this scoped - `if err := ...`

File: nakedret.go [hunk 4]
```diff
@@ -42,129 +48,97 @@ func main() {
...
-							return nil, err
-						}
-						files = append(files, f)
-					}
-
-				}
-			}
-		}
-	}
+			fmt.Println(reportFile)
```
Comment (alexkohler) [inline nakedret.go]: I don't think we want this `fmt.Println` here?

File: nakedret.go [hunk 5]
```diff
@@ -211,3 +194,76 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
...
+func checkRequestedFiles(dirName string, maxLength uint, shouldFix bool) {
+	path, err := filepath.Abs(dirName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	checkAllDirectories(path, maxLength, shouldFix)
+}
+
+func checkAllDirectories(path string, maxLength uint, shouldFix bool) {
```
Comment (alexkohler) [inline nakedret.go]: Shouln't we be bubbling these errors up? Currently, it looks like this will silently fail in some cases.

File: nakedret.go [hunk 6]
```diff
@@ -42,129 +48,97 @@ func main() {
...
+				formatted = b.Bytes()
+				log.Printf("got error while formatting: %v\n", err)
 			}
 
-			for _, pkg := range pkgs {
-				for _, f := range pkg.Files {
-					files = append(files, f)
-				}
-			}
+			ioutil.WriteFile(reportFile, formatted, os.ModeDevice)
```
Comment (alexkohler) [inline nakedret.go]: Why do we need `ModeDevice` for this write?

File: examples/example.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package examples
```
Comment (alexkohler) [inline examples/example.go]: Might be more appropriate to put this into a `testdata` directory

File: nakedret.go (hunk 6 above)
Comment (joshprzybyszewski-wf) [reply nakedret.go]: I was following some example; but I can't seem to find it. I'll use `0644` as in the go by example: https://gobyexample.com/writing-files

File: nakedret.go (hunk 5 above)
Comment (joshprzybyszewski-wf) [reply nakedret.go]: I chose to log errors in this func as a means of alerting rather than bubbling up. But there is a few opportunities for improvement.

**Comment by joshprzybyszewski-wf** [comment]
@alexkohler sorry for the delayed response -- should be ready for review again

**Comment by alexkohler** [comment]
Sorry for letting this fall by the wayside :disappointed: A lot of the fixes here ended up getting fixed in #25, so going to close this out.

**Pull request closed without merging**